### PR TITLE
[dagster-dbt] use Manifest.from_dict

### PR DESCRIPTION
**Pull request by alangenfeld**
replace the tricksy dict shim based `Manifest` with one built from the parsed json

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19643** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19643?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19643?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/sample_manifest.json
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/sample_manifest.json:1]: I just copied the latest `python_modules/libraries/dagster-dbt/dagster_dbt_tests/dagster_dbt_test_project/manifest.json` to here to get this to pass

**Review by rexledesma — APPROVED** [review]
✌🏼

**Review by OwenKephart — APPROVED** [review]
note comment, but very happy to get that blight out of the codebase

**Pull request merged**